### DISCUSSION TITLE
Set background-color when requesting asset thumbnails using fit=contain

### DISCUSSION THREAD

**Discussion by HitomiTenshi** [directus/directus#7261] [Feature Requests]
It would be nice if the `background` property of the [sharp resize API](https://sharp.pixelplumbing.com/api-resize#resize) could be exposed as well, otherwise resized thumbnails with the fit=contain setting always result with **black** letterboxes.

I was thinking about posting this as a bug report but decided to start a feature request instead since this isn't really a bug.

### Using hex values (input type: string):
`?fit=contain&background=%23fafafa`

### Using rgba values (input type: object / json):
`?fit=contain&background={ "r": 255, "g": 255, "b": 255, "alpha": 0.5 }`

`sharp` supports both string and object type inputs, since it's actually using the [color npm package](https://www.npmjs.com/package/color) under the hood.

PS: I noticed the [assets documentation](https://docs.directus.io/reference/api/assets/#requesting-a-thumbnail) says that the full sharp API is being exposed through the `transform` property but the example below shows it as transform**s**. I tested it and it seems the parameter with the **s** at the end seems to be the correct name. Should probably be corrected.

**Comment by rijkvanzanten** [comment]
Heya! The transform parameter you mentioned is meant for these more specific use cases. You can use fit=inside to get the contain-to-aspect-ratio without the letter boxing

**Reply by HitomiTenshi** [reply]
The issue is that the `background` option is part of the `resize` operation, which already gets internally used when specifying `fit=contain`, `width`, `height`, etc.

If I try to override the black background like this:

`?fit=contain&width=300&height=380&transforms=[["resize", {"background": "%23fafafa"}]]`

it doesn't work. Nevertheless I managed to fix my issue using fit=inside and some additional CSS, though I would be interested in a way that would solve the background coloring issue for future cases.

**Reply by rijkvanzanten** [reply]
You could use `?transforms=[["resize", {"background": "#fafafa", "width": 300, "height": 380, "fit": "contain"}]]`

**Comment by hmaesta** [comment]
I knew I couldn't be the only one having a hard time about this 😅
After trying several keywords, I finally found this ~issue~ post and manage to achieve what I need.

### Thumbnails with `fit: contain` and transparent background

If you are creating a **Storage Asset Preset**, paste this in `Transforms`:

```js
[
    [
        "resize",
        {
            "fit": "contain",
            "background": "rgba(0,0,0,0)",
            "width": 100,   // Change it
            "height": 100,  // Change it

        }
    ]
]
```

###### Ps: You need to set all values even if you already did in the native form

**Reply by rijkvanzanten** [reply]
@alesvaupotic Would be good to add some more in-depth examples of these uses to the assets docs 🙂

**Reply by alesvaupotic** [reply]
Agree, adding to my tasks.

**Reply by kentkravitz** [reply]
Wow! Thanks! Please add this specifically to the examples.  I almost started hacking on code thinking it wasn't possible.  (or add background to the officially supported options)

**Reply by rijkvanzanten** [reply]
@kiwimonk Do you mean the examples for [the assets docs](https://docs.directus.io/reference/files/#requesting-a-thumbnail) or [the Files & Storage section here](https://docs.directus.io/configuration/project-settings/#files-storage)?

**Comment by alesvaupotic** [comment]
Updated the docs for the plural `transforms` for now, will dig deeper and produce some more examples.

**Comment by M-GRV** [comment]
Hey!

Trying to apply this to Directus 10 and can't seem to figure it out. Assuming there was a major change to how this was handled between 8 & 9?

All my logos are looking like this. I have tried to replace the key as well used for the card views, but that didnt help. Just need to figure out how to change the background color for transparent PNG thumbnails. 
![image](https://github.com/directus/directus/assets/130182174/555f46f8-fb15-4e96-9d9f-c75447354abb)

**Reply by M-GRV** [reply]
I tried directly editing the key used in the card.vue component but that didn't work. 

Is there a way to specify this? Or maybe use "inside" as a card view?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by M-GRV** [reply]
I'm having this issue and found this to reference from... I posted in discord and here. Do you want me to open an issue instead? I don't mind, but found it to be redundant if the issue still exists.

**Reply by br41nslug** [reply]
This was a feature request not an open issue. If you're running into a bug please do open an issue for that @M-GRV

**Reply by M-GRV** [reply]
Issue created: [18578](https://github.com/directus/directus/issues/18578)
